src/main/scala/ai/privado/languageEngine/ruby/passes/download/ExternalDependenciesPass.scala

```diff
@@ -0,0 +1,82 @@
+package ai.privado.languageEngine.ruby.passes.download
+
+import ai.privado.languageEngine.ruby.cache.PackageTable
+import ai.privado.languageEngine.ruby.download.JRubyBasedParser
+import better.files.File
+import io.shiftleft.codepropertygraph.Cpg
+import io.shiftleft.passes.ConcurrentWriterCpgPass
+import org.jruby.Ruby
+import org.jruby.ast.{Colon2Node, DefnNode, Node, NodeType, ReturnNode}
+
+import scala.collection.mutable.ListBuffer
+import scala.jdk.CollectionConverters.CollectionHasAsScala
+
+class ExternalDependenciesPass(cpg: Cpg, tempExtDir: String, packageTable: PackageTable, inputPath: String)
+    extends ConcurrentWriterCpgPass[String](cpg) {
+  private val rubyInstance = Ruby.getGlobalRuntime()
+
+  override def generateParts(): Array[String] =
+    getRubyDependenciesFile(inputPath) ++ getRubyDependenciesFile(tempExtDir)
+
+  override def runOnPart(diffGraph: DiffGraphBuilder, filePath: String): Unit = {
+    val moduleName = resolveModuleNameFromPath(filePath)
+    try {
+      processRubyDependencyFile(filePath, moduleName)
+    } catch {
+      case ex: Exception =>
+        println(s"Error while parsing $moduleName module File ($filePath): ${ex.getMessage}")
+    }
+  }
+
+  private def fetchMethodInfoFromNode(node: Node, currentNameSpace: ListBuffer[String], moduleName: String): Unit = {
+    if (node != null) {
```

Comment:
Add exception handling here, as this may throw exception in unexpected scenarios

Reply:
Yes already added at function call.